src/components/Board.jsx

```diff
@@ -22,6 +29,51 @@ const Board = ({ id }) => {
     const columnOrderArray = board.columnOrder
     const { columns } = board
 
+    // TODO, move this function into utils folder
+    const onDragEnd = async (result) => {
+        const { destination, source, draggableId } = result
+
+        if (!destination) {
+            return
+        }
+
+        if (destination.droppableId === source.droppableId && destination.index === source.index) {
+            return
+        }
+
+        // When task is moved within one column
+        if (destination.droppableId === source.droppableId) {
+            const column = columns.find((col) => col.id === source.droppableId)
+            const newTaskOrder = Array.from(column.taskOrder)
+
+            newTaskOrder.splice(source.index, 1)
+            newTaskOrder.splice(destination.index, 0, draggableId)
+
+            await changeTaskOrderForOneColumn({
+                variables: {
+                    orderArray: newTaskOrder,
+                    columnId: column.id,
+                },
+            })
+            // TODO, figure out if the cache can be updated here before the mutation is sent
+            // Now there is a lag when showing the new order of tasks in the client
+            // When reordering the tasks the cache is updated and the new order is shown
+            // only once the response arrives from the server, this needs to be fixed
+        }
+
+        // When task is moved into another column
+        if (destination.droppableId !== source.droppableId) {
+            const sourceColumn = columns.find((col) => col.id === source.droppableId)
+            const destinationColumn = columns.find((col) => col.id === destination.droppableId)
+            const newTaskOrderOfSourceColumn = Array.from(sourceColumn.taskOrder)
+            const newTaskOrderOfDestinationColumn = Array.from(destinationColumn.taskOrder)
+
+            newTaskOrderOfSourceColumn.splice(source.index, 1)
+            newTaskOrderOfDestinationColumn.splice(destination.index, 0, draggableId)
```

Comment:
In-place mutations may cause unexpected side-effects later.

Reply:
Don't see a problem here since a new array is created from the old task order array before we splice anything. So we are not touching the original data used in rendering the component